### DISCUSSION TITLE
Support freeze for non-project qmd files

### DISCUSSION THREAD

**Discussion by th0ger** [quarto-dev/quarto-cli#6562] [Q&A]
### Description

I would like for external contributors to execute content in their own environments/git repos.
They will store ipynb notebooks with data in source control

The notebooks are pushed into a central git quarto project that will combine, render (no execute) and publish the full project.

This should work for ipynb.
For qmd, I would have to use something like the freeze functionality. But it seems that freeze is only supported for quarto *projects*, and not when rendering single files.

Is there a way to execute stand-alone qmd-files, save the execution output, and copy it into a quarto project?

**Comment by mcanouil** [comment]
You need the `_quarto.yml` to make your project a "Quarto project" then use the freeze option or the fact that Jupyter Notebook cells are not by default executed by Quarto, see <https://quarto.org/docs/projects/code-execution.html#notebooks>.
Note that your collaborator will have to push the freeze files in addition to source files[^1].

What do you think is missing?

[^1]: this is the current behaviour of <https://github.com/quarto-dev/quarto-web>.

**Reply by th0ger** [reply]
I'll like to support both qmd and ipynb (etc.) so contributers can pick their favorite.

Ipynb are simple and familiar, but has a weak point: The executed cells are included in the source code, which makes diff/review/merge hard.

Qmd is very nice, in that your .qmd is not modified when executed, but content can be stored in _freeze.
But it seems overkill to create a quarto project to be able to freeze a single qmd.
Also, these external quarto projects needs to be of the same type (website, blog) so the _freeze structure will fit into the target project?

**Reply by mcanouil** [reply]
> I'll like to support both qmd and ipynb (etc.) so contributers can pick their favorite.

Quarto supports both, there is nothing to do in particular for this.

> Ipynb are simple and familiar, but has a weak point: The executed cells are included in the source code, which makes diff/review/merge hard.

Indeed, but that's not something Quarto can do something about.

> But it seems overkill to create a quarto project to be able to freeze a single qmd.

Overkill to create `_quarto.yml` with `execute.freeze: auto/true/false`?
A Quarto project is simply defined by the presence of `_quarto.yml`. it does not have to be a book, a website, manuscript, etc. _i.e._, `quarto create project default myproject`.

> Also, these external quarto projects

What are external projects exactly?

Finally, are your contributors not cloning/forking (making a PR) the repository before working on it and contributing to it?

Additionally, I suggest to look for the caching feature, see <https://quarto.org/docs/computations/caching.html>.
